### ISSUE TITLE
Flow and es6 parse error

### ISSUE THREAD

**Issue by johnsonw** [jscs-dev/node-jscs#2084]
I'm running into a problem running jscs 2.8.0 when using es6 and flow. Take the following simple function:

```
(a:number, b:number):void => {

};
```

Running jscs on this will yield the following:

```
parseError: undefined at ./test.js :
     1 |(a:number, b:number):void => {
----------------------------^
     2 |
     3 |};
```

Now, let's remove one of the args:

```
(a:number):void => {

};
```

Running jscs on this function will pass. Let's add the arg back and remove the return type:

```
(a:number, b:number) => {

};
```

This also passes. So it seems that jscs fails when there are multiple arguments with specified types and a return type defined. Any idea what could be causing this?

**Comment by wayneseymour** [comment]
THIS IS DRIVING ME NUTS...please let me transpile in peace!  :8ball:

**Comment by hzoo** [comment]
If you get a parseError, it is due to an error in the parser not jscs itself. If you are using flow then it's using babel so its a babel 5 issue. It looks like it's fixed in babel 6 so we'll need to update babel-jscs to use babel 6 to do so.

**Comment by johnsonw** [comment]
any updates on this?

**Comment by hzoo** [comment]
This should be fixed in 3.0
